### PR TITLE
[WIP] Added Throttle Function

### PR DESCRIPTION
**Pull request by pon**
## What

Created initial version of `throttle` function. It rate limits the number of requests in a given time frame based on a provided key.

## Why

Rate limiting by key in Node using Redis.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -25,4 +26,106 @@ var Throttler = function (options) {
...
+
+/**
+ * Throttle execution based on the provided key
+ * @param {String} id
+ * @return {Promise}
+ */
+Throttler.prototype.throttle = function (id) {
+  var self = this;
+  return self.get(id)
+  .then(function (stats) {
```
Comment (robinjoseph08) [inline lib/index.js]: Won't the fact the this isn't atomic cause people to potentially go over their rate limit?

Comment (robinjoseph08) [inline lib/index.js]: Talked IRL - probably better to allow a few to go over the limit than to force some sort of locking mechanism whic would slow things down considerably.

@pon maybe this is worth noting in the README though

Comment (pon) [inline lib/index.js]: Alright sounds good

Comment (pon) [inline lib/index.js]: Actually - we should re-write like you said to kick off with an `incr` instead of a `get`

**Comment by pon** [comment]
This needs quite a bit more thought - closing

**Pull request closed without merging**